### PR TITLE
[mail-composer] fix adding attachment in bare

### PR DESCRIPTION
**Pull request by cruzach**
# Why

In the bare workflow, if you try to add an email attachment on Android you'll get 
```
<file-path> exposed beyond app through ClipData.Item.getUri()
```
This will resolve that issue

# How

Change provider name to the one defined in `AndroidManifest.xml` - https://github.com/expo/expo/blob/master/packages/expo-mail-composer/android/src/main/AndroidManifest.xml#L7

# Test Plan

Tested in a bare app and the expo client:
```
async function sendMail() {
  let fileUri = FileSystem.documentDirectory + "data.json";
  let data = { "this is": "mydata" };
  await FileSystem.writeAsStringAsync(fileUri, JSON.stringify(data), {
    encoding: FileSystem.EncodingType.UTF8,
  });
  await MailComposer.composeAsync({
    attachments: [fileUri],
    subject: "Hello",
    recipients: ["testRecipient@domain.com"],
  });
}
```

### CODE REVIEW

**Review by sjchmiela — APPROVED** [review]
Looks safe to me, but I would also like to ask @mczernek for an opinion. 👍

File: packages/expo-mail-composer/android/src/main/java/expo/modules/mailcomposer/MailComposerModule.java [hunk 1]
```diff
@@ -150,7 +150,7 @@ public void onHostDestroy() {
   private Uri contentUriFromFile(File file) {
     try {
       Application application = mModuleRegistry.getModule(ActivityProvider.class).getCurrentActivity().getApplication();
-      return FileProvider.getUriForFile(application, application.getPackageName() + ".provider", file);
+      return FileProvider.getUriForFile(application, application.getPackageName() + ".MailComposerFileProvider", file);
```
Comment (mczernek) [inline packages/expo-mail-composer/android/src/main/java/expo/modules/mailcomposer/MailComposerModule.java:153]: In general, authorities are named in package-like style. So I would go with `.mailcomposer.provider` both here and in `AndroidManifest.xml`

But of course, that should fix the issue, so feel free to merge as it is.

**Review by mczernek — APPROVED** [review]
Great work, just one quick suggestion.

File: packages/expo-mail-composer/android/src/main/java/expo/modules/mailcomposer/MailComposerModule.java (hunk 1 above)
Comment (cruzach) [reply packages/expo-mail-composer/android/src/main/java/expo/modules/mailcomposer/MailComposerModule.java:153]: ah okay, if that's the precedent then I agree we should do that. Do you think the same changes should be made to other packages using authorities the same way? (e.g. [expo-sharing](https://github.com/expo/expo/tree/master/packages/expo-sharing))

Comment (mczernek) [reply packages/expo-mail-composer/android/src/main/java/expo/modules/mailcomposer/MailComposerModule.java:153]: Thanks a lot for that!

I am a little worried about backward compatibility. I am almost (!) sure it won't break it, but would rather test i before. So maybe it is better to leave it as it is, than changed without testing.

**Pull request merged**